lib/src/user.dart

```diff
@@ -199,4 +283,39 @@ extension UserInternal on User {
   UserHandle get handle => _handle;
 
   static User create(UserHandle handle, [App? app]) => User._(handle, app);
+
+  static ApiKey createApiKey(ObjectId id, String name, String? value, bool isEnabled) => ApiKey._(id, name, value, isEnabled);
+}
+
+extension on Future<void> {
+  Future<void> handle404({ObjectId? id}) async {
+    try {
+      await this;
```

Comment:
I think this is different than my suggestion. What bothers me is we are awaiting the future here instead of just attaching an error handler to it. 
So if we didn't need to handle 404 here we would be returning an unawaited Future. But now we do await before returning it.
```dart
return realmCore.deleteApiKey(_user, objectId);
```
versus
```dart
return await realmCore.deleteApiKey(_user, objectId);
```

Reply:
I suggest we do with my previous code suggestion where we attach onError handlers and still return the Future unawaited

Reply:
extension on Future<ApiKey>

Reply:
Maybe I am missing something, but those seem functionally the same, why does it matter whether we use await or `onError`?

Reply:
https://dart.dev/guides/language/effective-dart/usage#prefer-asyncawait-over-using-raw-futures

Reply:
The difference is obvious . if you `await` you wait for the result of the future. If you don't await, you exit the method returning a future which can be awaited or chained even more before being awaited. If you notice we don't do any awaits in the SDK we return the futures unawaited in all other places. 
If you compare my suggestions with this one you will notice how clean my suggestion is. It is just chaining multiple futures and returning the last without awaiting it.

Reply:
Your suggestion is functionally the same as the code that uses async/await. Whether it's cleaner or not is a matter of personal preference, but the compiled code should be virtually the same.

```dart
Future<void> doSomething() { ... }

Future<void> doSomethingWithAwait() {
  try {
    await doSomething();
  } catch (e) {
    print(e);
  }
}

Future<void> doSomethingWithOnError() {
  return doSomething().onError((e) {
    print(e);
  });
}
```

These two are exactly the same. They will return a future that will either complete as soon as `doSomething` completes or will print a message when `doSomething` fails.

Reply:
There is a subtle difference between the two versions. In the first one you wait for the result and in the second you don't. We are using the second version in the SDK. What I am suggesting is to keep the consistency.

Reply:
For the fun of it I tried to decompile the generated code for the two functions. 

I only now how to do it for jit snapshots, and it would be better to compare the aot snapshots, but for the jit snapshots there is actually a difference in the amount of code generated, the `doSomethingWithOnError` being shorter.

I'm not sure if it has any impact on performance, or if it even holds true for aot compiled code (I'm trying to figure out how they do that - I know the dart sdk team can), but anyway.

As I think this is an interesting question, I'll try to clarify with the Dart SDK team.

For now, I don't think we should deviate from the [effective dart](https://dart.dev/guides/language/effective-dart/usage#prefer-asyncawait-over-using-raw-futures) recommendations, but @blagoev may have a point here, so I reserve the right to change my opinion later :-)

Reply:
BTW, I asked for how to decompile AOT code here: https://github.com/dart-lang/sdk/issues/50059

Reply:
> In the first one you wait for the result and in the second you don't

I'm not sure what that means. I dropped a note in the roundtable to discuss it as it'll probably be faster.

Reply:
Okay, we decided to prefer `await` when we need to do continuations.